src/services/reviews.js

```diff
@@ -0,0 +1,362 @@
+				{ id: resourceId, organization_id: resourceOrgId },
+				_.omit(updateObj, ['notes'])
+			)
+
+			await reviewsQueries.update({ id: reviewId, organization_id: reviewOrgId }, updateObj)
+
+			// Add or update comments
+			if (comments) {
+				await handleComments(comments, resourceId, loggedInUserId)
+			}
+
+			const message =
+				bodyData.status === common.REVIEW_STATUS_REJECTED ? 'REVIEW_REJECTED' : 'REVIEW_REJECTED_AND_REPORTED'
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message,
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async handleChangesRequested(
+		reviewId,
+		bodyData,
+		resourceId,
+		loggedInUserId,
+		comments = [],
+		organization_id
+	) {
+		try {
+			//update reviews table
+			await reviewsQueries.update({ id: reviewId, organization_id: organization_id }, { status: bodyData.status })
+
+			// Add or update comments
+			if (comments) {
+				await handleComments(comments, resourceId, loggedInUserId)
+			}
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'REVIEW_CHANGES_REQUESTED',
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	static async handleApproval(
+		reviewId,
+		resourceId,
+		loggedInUserId,
+		comments = [],
+		minApproval,
+		resourceOrgId,
+		reviewOrgId
+	) {
+		try {
+			// handle approval of a review
+			let publishResource = false
+			await reviewsQueries.update(
+				{ id: reviewId, organization_id: reviewOrgId },
+				{ status: common.RESOURCE_STATUS_APPROVED }
+			)
+
+			// Add or update comments
+			if (comments) {
+				await handleComments(comments, resourceId, loggedInUserId)
+			}
+
+			//check the no of approvals meets
+			const reviewsApproved = await reviewsQueries.countDistinct({
+				resource_id: resourceId,
+				status: common.REVIEW_STATUS_APPROVED,
+			})
+
+			if (minApproval <= reviewsApproved + 1) {
+				await resourceQueries.updateOne(
+					{ id: resourceId, organization_id: resourceOrgId },
+					{ status: common.RESOURCE_STATUS_PUBLISHED }
+				)
+				publishResource = true
+			}
+
+			return publishResource
+		} catch (error) {
+			throw error
+		}
+	}
+}
+
+const _notAllowedStatusForReview = [
+	common.RESOURCE_STATUS_REJECTED,
+	common.RESOURCE_STATUS_REJECTED_AND_REPORTED,
+	common.RESOURCE_STATUS_PUBLISHED,
+]
+
+const _notAllowedReviewStatus = [
+	common.RESOURCE_STATUS_REJECTED,
+	common.REVIEW_STATUS_REQUESTED_FOR_CHANGES,
+	common.REVIEW_STATUS_CHANGES_UPDATED,
+	common.REVIEW_STATUS_INPROGRESS,
+	common.REVIEW_STATUS_REJECTED_AND_REPORTED,
+]
+
+async function handleComments(comments, resourceId, loggedInUserId) {
+	try {
+		// Normalize comments to an array if it's a single object
+		if (!Array.isArray(comments)) {
+			comments = [comments]
+		}
+		// handle adding or updating comments
+		for (const comment of comments) {
+			if (comment.id) {
+				await commentQueries.updateOne({ id: comment.id, resource_id: resourceId }, _.omit(comment, ['id']))
```

Comment:
We can update only draft comments right? Should we check the status also here? @priyanka-TL

Reply:
no, they can resolve it also that is not a separate API all